connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/DelegatingClassLoader.java

```diff
@@ -408,4 +407,18 @@ protected void scan(URL url) {
             }
         }
     }
+
+    @Override
+    public URL getResource(String name) {
```

Comment:
Are we happy with this solution? Are we sure it wouldn't ever make sense to load resources from the classpath? As an example of where I could see this going wrong, we include Kafka version info as a resource, which is what AppInfoParser loads to get version info. Right now we'd be ok because the included connectors/transforms are on the classpath and will still be able to get at that info, but a plugin would not. I know Confluent ones are ok because they include their own metadata files to track versions and I doubt any other connectors are using that info, but do we feel comfortable just cutting off access completely?

Another way of addressing this would be to require REST extension plugins to use the plugin path functionality. We want to encourage people that direction anyway, and as we support this for other plugin types we could just require that as you move over to that instead of classpath scanning, you should make sure you deploy using plugin.path.

Reply:
The downside of not doing this is if that any plugin ( that is designed to work with ServiceLoader mechanism) available in the classpath, gets listed in all the plugin class loaders. Another approach to solving this problem would be to cut the circuit only for known MANIFEST service files in the PluginClassLoader. So, we will return null only for "META-INF/services/org.apache.kafka.connect.rest.ConnectRestExtension" or the ConfigProvider Manifest. This would mean any new plugins would need to be added to this set. Let me know your thoughts.

Reply:
Isolation + ServiceLoader, I really need to start looking for systems that do that :). How about something like

`        if (serviceLoaderManifestForPlugin(name)) {
            // Default implementation of getResources searches the parent class loader and also its own URL paths. This will enable the
            // PluginClassLoader to limit its resource search to only its own URL paths.
            return null;
        } else {
            return super.getResources(name);
        }`

Reply:
Yeah, I think the solution that limits to just the manifests we know shouldn't be loaded from the main classloader is a reasonable compromise. I was just curious because I figured most pluggable systems would eventually want classpath isolation and it seemed like they would all hit it. I guess most are just praying that they don't see conflicts.

Reply:
Yeah, I was a little bit optimistic here. The good thing as you both mention is that we don't have to blacklist anything. We just need to whitelist the plugins we are interested in not delegating to the parent. That's all.